### ISSUE TITLE
Error in { : task 1 failed - "non-conformable arguments"

### ISSUE THREAD

**Issue by julianmarenz** [asheshrambachan/HonestDiD#9]
Hello!

First, thank you for your great work and the package. I am experiencing some issues pertaining to non-conformable arguments. 

I am estimating a fixed-effects regression with entity and time fixed effects (with ```plm```, using a within estimation, individual fixed effects and manually added time dummies). Subsequently, I adjust standard errors and cluster by group. I then wanted to employ the tests that you have described in the vignette.

Replicating the vignette works fine for me. However, when I try to apply the same steps to my own data, estimation fails. The structure of my data appears to be very similar, albeit a difference in the number of pre periods and more coefficients. Could the failure have something to do with the structure of sigma? I have posted the console output below. Hope this is an appropriate place to flag my issue.

```
> HKLO_numPrePeriods
[1] 5
> HKLO_numPostPeriods
[1] 4
> HKLO_l_vec
     [,1]
[1,]    1
[2,]    0
[3,]    0
[4,]    0
> beta_1
 [1]  0.025639471  0.012862650  0.031613393  1.214880752  0.043718345  0.050889448  0.020796999 -0.006101869  0.077520292
[10]  0.013092666  0.012840537 -0.013910122
> sigma_1
               [,1]          [,2]          [,3]          [,4]          [,5]          [,6]          [,7]          [,8]
 [1,]  5.960917e-05 -3.948848e-05 -1.861408e-05  0.0001701266 -1.272809e-06 -2.390593e-06 -2.523283e-06 -5.074482e-06
 [2,] -3.948848e-05  2.676426e-04  5.116043e-06  0.0001324583 -3.985026e-06 -8.084589e-06 -1.329489e-05 -1.825596e-05
 [3,] -1.861408e-05  5.116043e-06  1.492741e-04 -0.0005439702 -1.200232e-05 -2.171110e-05 -2.940021e-05 -1.481694e-05
 [4,]  1.701266e-04  1.324583e-04 -5.439702e-04  0.0193808492 -3.796138e-04 -7.724937e-04 -1.180764e-03 -1.706985e-03
 [5,] -1.272809e-06 -3.985026e-06 -1.200232e-05 -0.0003796138  1.317267e-05  2.476562e-05  3.611966e-05  4.652359e-05
 [6,] -2.390593e-06 -8.084589e-06 -2.171110e-05 -0.0007724937  2.476562e-05  4.916163e-05  7.186124e-05  9.315773e-05
 [7,] -2.523283e-06 -1.329489e-05 -2.940021e-05 -0.0011807636  3.611966e-05  7.186124e-05  1.074581e-04  1.399805e-04
 [8,] -5.074482e-06 -1.825596e-05 -1.481694e-05 -0.0017069845  4.652359e-05  9.315773e-05  1.399805e-04  1.879543e-04
 [9,] -2.350533e-05 -1.105146e-05 -1.147658e-05 -0.0021857571  5.920348e-05  1.177208e-04  1.751695e-04  2.352674e-04
[10,] -2.425382e-05 -1.738534e-05 -1.354381e-05 -0.0026442084  7.036862e-05  1.403867e-04  2.100267e-04  2.826409e-04
[11,] -2.453522e-05 -2.176817e-05 -1.089759e-05 -0.0030996508  8.141667e-05  1.623130e-04  2.427686e-04  3.278943e-04
[12,] -2.663931e-05 -2.742002e-05 -7.637740e-06 -0.0035650541  9.220028e-05  1.841173e-04  2.758906e-04  3.736986e-04
               [,9]         [,10]         [,11]         [,12]
 [1,] -2.350533e-05 -2.425382e-05 -2.453522e-05 -2.663931e-05
 [2,] -1.105146e-05 -1.738534e-05 -2.176817e-05 -2.742002e-05
 [3,] -1.147658e-05 -1.354381e-05 -1.089759e-05 -7.637740e-06
 [4,] -2.185757e-03 -2.644208e-03 -3.099651e-03 -3.565054e-03
 [5,]  5.920348e-05  7.036862e-05  8.141667e-05  9.220028e-05
 [6,]  1.177208e-04  1.403867e-04  1.623130e-04  1.841173e-04
 [7,]  1.751695e-04  2.100267e-04  2.427686e-04  2.758906e-04
 [8,]  2.352674e-04  2.826409e-04  3.278943e-04  3.736986e-04
 [9,]  3.046934e-04  3.627664e-04  4.205758e-04  4.782563e-04
[10,]  3.627664e-04  4.357224e-04  5.045653e-04  5.748301e-04
[11,]  4.205758e-04  5.045653e-04  5.872913e-04  6.690070e-04
[12,]  4.782563e-04  5.748301e-04  6.690070e-04  7.644673e-04
> HKLO_DeltaSDRM_RobustResults = createSensitivityResults_relativeMagnitudes(betahat = beta_1, 
+                                                                            sigma = sigma_1,
+                                                                            bound = "deviation from linear trend",
+                                                                            numPrePeriods = HKLO_numPrePeriods, 
+                                                                            numPostPeriods = HKLO_numPostPeriods,
+                                                                            l_vec = HKLO_l_vec, 
+                                                                            Mbarvec = seq(from = 0, to = 2, by = 0.5), 
+                                                                            gridPoints = 100, grid.lb = -1, grid.ub = 1)
Error in { : task 1 failed - "non-conformable arguments"
```

**Comment by asheshrambachan** [comment]
@julianmarenz `beta_1` looks like it's a vector with 12 values, but you specified that there are only 5 pre-periods and 4 post-periods. I suspect that is creating the issue.

**Comment by julianmarenz** [comment]
@asheshrambachan thanks for the reply! Do I understand you correctly then, that we need at least as many periods as we have coefficients? Relatedly, if we use e.g. entity or time fixed effects, would the number of pre/post periods necessary then depend on whether we include dummies (increasing the number of variables substantially) or demean our values?

**Comment by asheshrambachan** [comment]
@julianmarenz `beta` should only include the pre/post-period coefficients and `sigma` should only include the elements of the VCV matrix associated with those estimated coefficients. 

For example, if your event study regression looks something like 
![Screen Shot 2022-02-06 at 12 18 57 PM](https://user-images.githubusercontent.com/29416461/152692644-da9087c4-c966-455f-acfa-c76826853b36.png)
from the vignette, you would not include the additional "controls" or fixed effects in the input `beta` and `sigma`.  The input `beta` only consists of the estimated event study coefficients (beta in the regression above) and `sigma` only consists of the elements of the VCV matrix associated with the estimated event study coefficients.

**Comment by julianmarenz** [comment]
@asheshrambachan of course! I should have been more attentive reading and noticing that the beta vector is only including the event study coefficients. Thank you for clarifying. Works like a charm now.

As an side, when I did run the ```createSensitivityResults_relativeMagnitudes``` function I needed to install the two packages ```mvtnorm``` and ```Rglpk``` first. Seems like (at least on my machine) these are dependencies that were not automatically installed.
